configs/main.go

```diff
@@ -48,24 +49,46 @@ func (c *Configs) unmarshalConfig(config *Config, data interface{}) error {
 }
 
 func (c *Configs) marshalConfig(config *Config, cfg interface{}) error {
-	reflectCfg := reflect.ValueOf(cfg)
-	for i := 0; i < reflectCfg.NumField(); i++ {
-		k := reflectCfg.Type().Field(i).Name
-		v := reflectCfg.Field(i).Interface()
-		config.viper.Set(k, v)
-	}
+	// reflectCfg := reflect.ValueOf(cfg)
+	// for i := 0; i < reflectCfg.NumField(); i++ {
+	// 	k := reflectCfg.Type().Field(i).Name
+	// 	v := reflectCfg.Field(i).Interface()
+	// 	config.viper.Set(k, v)
+	// }
+
+	// err = config.viper.WriteConfig()
 
 	err := c.CreatePathIfNotExist(config.configPath)
 	if err != nil {
 		return err
 	}
 
-	err = config.viper.WriteConfig()
+	jsonString, err := json.MarshalIndent(cfg, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(config.configPath, jsonString, os.ModePerm)
+	if err != nil {
+		return err
+	}
```

Comment:
If we're using viper still, shouldn't we be doing this via viper? I'm not a huge fan of having two mechanisms to do io

Reply:
Yeah I agree it isn't ideal to go around viper and just write the json manually. But I couldn't figure out how to just write an entire config file with viper and doing the recursive reflect stuff seems more complicated than it needs to be. I'll do some digging in the viper docs and try and find a better solution.